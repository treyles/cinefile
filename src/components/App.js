import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Header from './Header';
import Library from './Library';
import Discover from './Discover';
import Search from './Search';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      library: []
    };

    this.addToLibrary = this.addToLibrary.bind(this);
    this.removeFromLibrary = this.removeFromLibrary.bind(this);
  }

  // TODO: rename 'handle'
  addToLibrary(media) {
    this.setState({
      library: [media].concat(this.state.library)
    });
  }

  removeFromLibrary(media) {
    const { library } = this.state;
    const mediaIndex = library.indexOf(media);

    if (mediaIndex > -1) {
      this.setState({
        library: library
          .slice(0, mediaIndex)
          .concat(library.slice(mediaIndex + 1))
      });
    }
  }

  render() {
    const { library } = this.state;

    return (
      <BrowserRouter>
        <div className="app">
          <Header count={library.length} />
          <Switch>
            <Route
              exact
              path="/"
              render={() => (
                <Library
                  library={library}
                  removeFromLibrary={this.removeFromLibrary}
                />
              )}
            />
            <Route
              path="/discover"
              render={() => <Discover library={library} />}
            />
            <Route
              path="/search"
              render={() => <Search addToLibrary={this.addToLibrary} />}
            />
            <Route
              render={() => <p className="not-found">Not Found!</p>}
            />
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
}
