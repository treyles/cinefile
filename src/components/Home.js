import React from 'react';
import { NavLink } from 'react-router-dom';
import ReactModal from 'react-modal';
import base from '../utils/base';
import firebase from 'firebase';
import heroPreview from '../assets/hero-preview.mp4';
import heroPoster from '../assets/hero-preview-poster.jpg';
import mobilePreview from '../assets/mobile-preview.jpg';
// import mobilePoster from '../assets/mobile-preview-poster.jpg';
import SignIn from './SignIn';
// import mobilePoster from '../assets/mobile-preview-poster.jpg';
// import mobile from '../assets/mobile.png';
// import { Link } from 'react-router-dom';

export default class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showSignIn: false
    };

    this.handleSignInModal = this.handleSignInModal.bind(this);
  }

  componentWillMount() {
    document.body.style.backgroundColor = '#0883df';
  }

  componentWillUnmount() {
    document.body.style.backgroundColor = null;
  }

  handleSignInModal() {
    this.setState({
      showSignIn: !this.state.showSignIn
    });
  }

  render() {
    const { showSignIn } = this.state;
    const about = 'https://github.com/treyles/cinefile';

    return (
      <div className="home">
        <div className="home-header">
          <h1>Cinefile</h1>
          <ul>
            <li>
              <a href={about} target="blank">
                About
              </a>
            </li>
            <li>
              <button
                className="signin-btn"
                onClick={this.handleSignInModal}
              >
                Sign In
              </button>
            </li>
          </ul>
        </div>
        <div className="hero">
          <div className="hero-content">
            <h1 className="title">
              When you see something you want to watch later, put it in
              Cinefile.
            </h1>
            <p>
              Keep movies, tv shows, and documentaries from slipping off
              your radar with Cinefile — a minimal bookmarking and
              discovery app
            </p>
            <button
              className="get-started-btn"
              onClick={this.handleSignInModal}
            >
              Get Started, It's Free!
            </button>
          </div>
          <div className="hero-video-container">
            <div className="video-window" />
            <div className="video-content-container">
              <video
                className="hero-preview"
                poster={heroPoster}
                autoPlay
                loop
              >
                <source src={heroPreview} type="video/mp4" />
              </video>
            </div>
          </div>
        </div>
        <div className="mobile-section">
          <div className="mobile-content-container">
            <img src={mobilePreview} />
          </div>
          <div className="text-block">
            <h1 className="text-block-header">Mobile Friendly</h1>
            <p className="text-block-one">
              Always ready to save that tv show your friend just told you
              about.
            </p>
            <h1 className="text-block-header">Minimal Design</h1>
            <p>
              When we say minimal, we mean it. Cinefile provides a rating,
              year, short synopsis, director, lead actor, and trailer. It
              also links to it’s corresponding IMDB page if that doesn’t
              satisfy you.
            </p>
          </div>
        </div>
        <div className="footer">
          <button className="footer-btn" onClick={this.handleSignInModal}>
            Get Started!
          </button>
          <h2>MADE BY @TREYLES</h2>
        </div>
        <ReactModal
          isOpen={showSignIn}
          onRequestClose={this.handleSignInModal}
          className="sign-in-modal"
          overlayClassName="sign-in-overlay"
          ariaHideApp={false}
        >
          <SignIn handleSignInModal={this.handleSignInModal} />
        </ReactModal>
      </div>
    );
  }
}

/*
            <button
              className="get-started-btn"
              onClick={() => handleAuthorization()}
            >
*/

/*
export default class Home extends React.Component {
  render() {
    const { handleAuthorization } = this.props;
    return (
      <div className="home">
        <button onClick={() => handleAuthorization()}>
          <h3>Connect with Google</h3>
        </button>
      </div>
    );
  }
}
*/
