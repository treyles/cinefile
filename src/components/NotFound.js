import React from 'react';
import Icon from '../utils/Icon';
import MediaQuery from 'react-responsive';

export default function NotFound() {
  return (
    <div className="lobby not-found">
      <div className="empty-message">
        <div className="illustration">
          <MediaQuery minWidth={768}>
            <Icon icon="sad" width="218" height="100%" />
          </MediaQuery>
          <MediaQuery maxWidth={768}>
            <Icon icon="sadMobile" width="137" height="100%" />
          </MediaQuery>
        </div>
        <h1>Page Not Found!</h1>
        <h2>The page you are looking for doesn&apos;t exist</h2>
      </div>
    </div>
  );
}
