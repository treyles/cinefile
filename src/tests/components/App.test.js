import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import App from '../../components/App';
import Home from '../../components/Home';
import NotFound from '../../components/NotFound';


describe('App', () => {
  let component;

  beforeEach(() => {
      component = shallow(<App />);
  })

  it('should render correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('should render render `Home` if currentUser is null', () => {
    const mountedComponent = mount(
      <MemoryRouter initialEntries={[ '/' ]}>
        <App/>
      </MemoryRouter>
    )

    expect(mountedComponent.find(Home)).toHaveLength(1)
  })

  it('should render `NotFound` if visiting unknown path', () => {
    const mountedComponent = mount(
      <MemoryRouter initialEntries={[ '/unknownpath' ]}>
        <App/>
      </MemoryRouter>
    )
    
    expect(mountedComponent.find(NotFound)).toHaveLength(1)
  })

  it('should push new media with `addToLibrary`', () => {
    const media = { one: 'one' }

    component.instance().addToLibrary(media)
    expect(component.state().library).toHaveLength(1)
  })

  it('should push new media to top (index 0) with `addToLibrary`', () => {
    component.setState({
      library: [{ one: 'one'}]
    })

    const media = { two: 'two' }

    component.instance().addToLibrary(media)
    expect(component.state().library[0]).toEqual(media)
  })

  it('should remove new media with `removeFromLibrary`', () => {
    const media = { one: 'one' }
    const mediaTwo = {two: 'two'}

    component.setState({
      library: [media, mediaTwo]
    })

    component.instance().removeFromLibrary(media)

    expect(component.state().library).toHaveLength(1)
    expect(component.state().library[0]).toEqual(mediaTwo)
  })

  it('should load recommended media on user request', () => {
    component.instance().addRecommended()
    expect(component.state().library).toHaveLength(14)

  })
})