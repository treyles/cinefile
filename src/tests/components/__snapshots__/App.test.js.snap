// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="app"
      >
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            path="/discover"
            render={[Function]}
          />
          <Route
            render={[Function]}
          />
        </Switch>
        <div
          className="alert false"
        >
          <h2 />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/discover"
              render={[Function]}
            />
            <Route
              render={[Function]}
            />
          </Switch>,
          <div
            className="alert false"
          >
            <h2 />
          </div>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                path="/discover"
                render={[Function]}
              />,
              <Route
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/discover",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2 />,
            "className": "alert false",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": false,
            },
            "ref": null,
            "rendered": false,
            "type": "h2",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="app"
        >
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/discover"
              render={[Function]}
            />
            <Route
              render={[Function]}
            />
          </Switch>
          <div
            className="alert false"
          >
            <h2 />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Switch>
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                path="/discover"
                render={[Function]}
              />
              <Route
                render={[Function]}
              />
            </Switch>,
            <div
              className="alert false"
            >
              <h2 />
            </div>,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
                />,
                <Route
                  path="/discover"
                  render={[Function]}
                />,
                <Route
                  render={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/discover",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2 />,
              "className": "alert false",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": false,
              },
              "ref": null,
              "rendered": false,
              "type": "h2",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
